backend/infrahub/core/diff/conflicts_identifier.py

```diff
@@ -0,0 +1,266 @@
+    ) -> list[DataConflict]:
+        conflicts: list[DataConflict] = []
+        node_schema = self.schema_manager.get_node_schema(
+            name=branch_node.kind, branch=self.diff_branch, duplicate=False
+        )
+        relationship_schema = node_schema.get_relationship(name=branch_relationship.name)
+        is_cardinality_one = relationship_schema.cardinality is RelationshipCardinality.ONE
+        if is_cardinality_one:
+            base_element = next(iter(base_relationship.relationships))
+            branch_element = next(iter(branch_relationship.relationships))
+            element_tuples = [(base_element, branch_element)]
+        else:
+            base_peer_id_map = {element.peer_id: element for element in base_relationship.relationships}
+            branch_peer_id_map = {element.peer_id: element for element in branch_relationship.relationships}
+            common_peer_ids = set(base_peer_id_map.keys()) & set(branch_peer_id_map.keys())
+            element_tuples = [(base_peer_id_map[peer_id], branch_peer_id_map[peer_id]) for peer_id in common_peer_ids]
+        for base_element, branch_element in element_tuples:
+            conflicts.extend(
+                self._add_relationship_conflicts_for_one_peer(
+                    branch_node=branch_node,
+                    branch_relationship=branch_relationship,
+                    base_element=base_element,
+                    branch_element=branch_element,
+                    is_cardinality_one=is_cardinality_one,
+                )
+            )
+        return conflicts
+
+    def _add_relationship_conflicts_for_one_peer(
+        self,
+        branch_node: EnrichedDiffNode,
+        branch_relationship: EnrichedDiffRelationship,
+        base_element: EnrichedDiffSingleRelationship,
+        branch_element: EnrichedDiffSingleRelationship,
+        is_cardinality_one: bool,
+    ) -> list[DataConflict]:
+        base_properties_by_type = {p.property_type: p for p in base_element.properties}
+        branch_properties_by_type = {p.property_type: p for p in branch_element.properties}
+        common_property_types = set(base_properties_by_type.keys()) & set(branch_properties_by_type.keys())
+        if not common_property_types:
+            return []
+        if is_cardinality_one:
+            path_type = PathType.RELATIONSHIP_ONE
+        else:
+            path_type = PathType.RELATIONSHIP_MANY
+        conflicts: list[DataConflict] = []
+        for property_type in common_property_types:
+            base_property = base_properties_by_type[property_type]
+            branch_property = branch_properties_by_type[property_type]
+            base_path = f"{ModifiedPathType.DATA.value}/{branch_node.uuid}/{branch_relationship.name}/peer"
+            if property_type in (DatabaseEdgeType.HAS_VALUE, DatabaseEdgeType.IS_RELATED):
+                change_type = f"{path_type.value}_value"
+            else:
+                change_type = f"{path_type.value}_property"
```

Comment:
This if/else clause could potentially be handled by having a change_type property on the DatabaseEdgeType object itself.

Reply:
I was thinking that `DatabaseEdgeType` could be used in other parts of the code and did not want to include diff/conflict-specific stuff on that class